### PR TITLE
ISSUE-208: Search API Twig extension, Map Views and Cleanup

### PR DESCRIPTION
**Pull request by DiegoPino**
# WIP (too tired to keep coding tonight... what is wrong with me?)

See #208 

So far: Does not return yet but query/facet is working fine. Need to parse the result to make it suitable for traversing/rendering in the template still.

The basic operation is
```Twig
{{ sbf_search_api('default_solr_index','Financing',[],{'digital_object_type':'Book'},['type_1']) }}
```

which in my own local will give me this (e.g) for the facets
```shell
result = {array} [3]
 search_api_solr_response = {array} [3]
 facet_set = {Solarium\Component\Result\FacetSet} [1]
 search_api_facets = {array} [1]
  facet:type_1 = {array} [5]
   0 = {array} [2]
    filter = ""Book""
    count = {int} 2
   1 = {array} [2]
    filter = ""Image""
    count = {int} 2
   2 = {array} [2]
    filter = ""Service""
    count = {int} 2
   3 = {array} [2]
    filter = ""Update""
    count = {int} 2
   4 = {array} [2]
    filter = ""Document""
    count = {int} 1
```

and for the actual DOC:

```Shell
resultItems = {array} [1]
 entity:node/5254:en = {Drupal\search_api\Item\Item} [14]
  index = {Drupal\search_api\Entity\Index} [36]
  itemId = "entity:node/5254:en"
  originalObject = null
  datasourceId = null
  datasource = null
  language = "en"
  fields = {array} [21]
   as_image_exif_mimetype = {Drupal\search_api\Item\Field} [22]
   author = {Drupal\search_api\Item\Field} [22]
   changed = {Drupal\search_api\Item\Field} [22]
   created = {Drupal\search_api\Item\Field} [22]
   digital_object_type = {Drupal\search_api\Item\Field} [22]
   label_1 = {Drupal\search_api\Item\Field} [22]
   label_2 = {Drupal\search_api\Item\Field} [22]
   name_1 = {Drupal\search_api\Item\Field} [22]
   nid = {Drupal\search_api\Item\Field} [22]
   node_grants = {Drupal\search_api\Item\Field} [22]
   rendered_item = {Drupal\search_api\Item\Field} [22]
   status = {Drupal\search_api\Item\Field} [22]
   sticky = {Drupal\search_api\Item\Field} [22]
   str_flatten_keys = {Drupal\search_api\Item\Field} [22]
   title = {Drupal\search_api\Item\Field} [22]
   title_autocomplete = {Drupal\search_api\Item\Field} [22]
   title_string = {Drupal\search_api\Item\Field} [22]
   title_unstemmed = {Drupal\search_api\Item\Field} [22]
   type = {Drupal\search_api\Item\Field} [22]
   type_1 = {Drupal\search_api\Item\Field} [22]
   uid = {Drupal\search_api\Item\Field} [22]
  fieldsExtracted = false
  excerpt = null
  score = {float} 5.0232134
  boost = {float} 1.0
  extraData = {array} [3]
  accessResults = {array} [0]
  logger = null
```

All is there. Wonder now if for this complex/advanced one I "should" return [] in case of exception but! Have a debug/option (user logged in?/certain permission) to show the why it failed ?

@patdunlavey @aksm @alliomeria  
(you might wonder why this thing... super super useful, you will someday need it!)

### CODE REVIEW

**Review by aksm — APPROVED** [review]

File: src/Form/MetadataDisplayForm.php [hunk 1]
```diff
@@ -143,6 +146,8 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
       // Do not set Form Errors if running a Preview Operation.
       if (isset($form_state->getTriggeringElement()['#type']) &&
         $form_state->getTriggeringElement()['#type'] == 'submit') {
+        // This is not showing correctly. Why is the message missing?
+        $this->messenger()->addError($message);
```
[the remark this answers is not in the document]
Comment (DiegoPino) [reply src/Form/MetadataDisplayForm.php:150]: Not sure why, but in the theme I can not see the actual error message, only the flagged element as errored. I think the issue is that also the Entity Validation kicks in afterwards (parent:: validation)

[the remark this answers is not in the document]
Comment (aksm) [reply src/Form/MetadataDisplayForm.php:150]: Hmm, just tried with the Archipelago base theme enabled for admin, and the error message is still showing as expected.

[the remark this answers is not in the document]
Comment (DiegoPino) [reply src/Form/MetadataDisplayForm.php:150]: Screenshot??

[the remark this answers is not in the document]
Comment (aksm) [reply src/Form/MetadataDisplayForm.php:150]: Screenshot from 2022-06-30 12-20-13

**Comment by DiegoPino** [comment]
@aksm thanks! Can you (last one) update the README.md with a small note saying how to install via crush the cite-rod libraries? Sorry!

**Pull request merged**